rbac/common/role/create_role.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2018 Contributors to Hyperledger Sawtooth
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+import logging
+from rbac.addressing import addresser
+from rbac.common import protobuf
+from rbac.common.protobuf.rbac_payload_pb2 import RBACPayload
+from rbac.common.manager.base_message import BaseMessage
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CreateRole(BaseMessage):
+    def __init__(self):
+        BaseMessage.__init__(self)
+
+    @property
+    def name(self):
+        return "role"
+
+    @property
+    def names(self):
+        return self.name + "_attributes"
+
+    @property
+    def message_type(self):
+        return RBACPayload.CREATE_ROLE
+
+    @property
+    def message_proto(self):
+        return protobuf.role_transaction_pb2.CreateRole
+
+    @property
+    def container_proto(self):
+        return protobuf.role_state_pb2.RoleAttributesContainer
+
+    @property
+    def state_proto(self):
+        return protobuf.role_state_pb2.Role
+
+    @property
+    def message_fields_not_in_state(self):
+        """Fields that are on the message but not stored on the state object"""
+        return ["owners", "admins"]
+
+    def address(self, object_id, target_id=None):
```

Comment:
This is another function that is copy-pasted in numerous places. Can it be consolidated into one location and only overridden when necessary?

Reply:
No, classes implement all properties and methods required by the abstract class.

Reply:
I understand that, but why require subclasses to implement methods when their core logic is identical? Is it not possible to implement shared logic in the base class and override only when necessary in the subclasses?